workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/registry.py

```diff
@@ -0,0 +1,23 @@
+from collections import defaultdict
+from typing import Any, Callable, Mapping
+
+import xarray as xr
+from toolz import curry
+
+
+class Registry:
+    def __init__(self, merge_func: Callable[[Mapping[str, xr.Dataset]], Any]):
```

Comment:
Nice job parameterizing the merge function! I would just call it "merge" since that is a good verb for a function name.